transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -572,21 +568,16 @@ private void processSelectedKeysOptimized(SelectionKey[] selectedKeys) {
             if (needsToSelectAgain) {
                 // null out entries in the array to allow to have it GC'ed once the Channel close
                 // See https://github.com/netty/netty/issues/2363
-                for (;;) {
-                    i++;
-                    if (selectedKeys[i] == null) {
-                        break;
-                    }
-                    selectedKeys[i] = null;
-                }
+                this.selectedKeys.reset(i + 1);
```

Comment:
right but the previous impl stopped iterating on the first non null element found presumably to avoid iterating the full set and burn less cpu cycles?